erc20/ds-token/README.md

```diff
@@ -0,0 +1,113 @@
+*2018-03-12*
+
+# Formal Verification of DSToken ERC20 Token Contract
+
+We present a formal verification of [DSToken ERC20][token] token contract.
+
+The DSToken ERC20 token is a high-profile ERC20 token contract written in Solidity implemented by [`DappHub`](https://dapphub.com/).
+
+We found that the DSToken ERC20 token deviates from the [ERC20-K] (and thus [ERC20-EVM]) specification as follows:
+
+* *Safe checks for overflow and underflow*: For both `transfer` and `transferFrom` it checks for underflow and for overflow, throwing an exception immediately.
+* *Stoppable modifier*: It uses the stoppable modifier from `ds-stop`, which if it is set to true by the owner of the token it throws exception in `transfer`, `transferFrom` and `approve`. 
+
+## Target Smart Contract
+
+The target contract of our formal verification is the following, where we took the Solidity source code from the Github repository,
+https://github.com/dapphub/ds-token/tree/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb,
+commit [`1a1a5b6`][version]:
+* [token.sol][token]
+* [base.sol][base]
+
+We formally verified the full functional correctness of the following ERC20 functions:
+
+* [`totalSupply`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/base.sol#L27) - inherited in DSToken from DSTokenBase 
+* [`balanceOf`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/base.sol#L30) - inherited in DSToken from DSTokenBase 
+* [`allowance`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/base.sol#L33) - inherited in DSToken from DSTokenBase 
+* [`approve`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/token.sol#L35) which extends [`approve`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/base.sol#L61) from DSTokenBase
+* [`transfer`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/token.sol#L27) which extends [`transfer`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/base.sol#L37) from DSTokenBase
+* [`transferFrom`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/token.sol#L30) which extends [`transferFrom`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/base.sol#L48) from DSTokenBase
+
+## Verification Artifacts
+
+### Solidity Source Code and Compiled EVM Bytecode
+
+We took the [token source code][token], [base token source code][base] and the additional libraries [ds-auth], [ds-math], [ds-stop] and [ds-note], and compiled it to the EVM bytecode using solc compiler (of the version `0.4.18+commit.9cf6e910.Linux.g++`).
```

Comment:
Use the contract names, (DSToken, DSTokenBase, DSAuth, DSMath, DSStop, and DSNote), instead of the file names.

Also, fix the missing links.